apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx

```diff
@@ -34,45 +46,60 @@ import {
 } from 'ui-patterns/multi-select'
 import { z } from 'zod'
 
-const formSchema = z.object({
-  dbSchema: z.array(z.string()),
-  dbExtraSearchPath: z.string(),
-  maxRows: z.number().max(1000000, "Can't be more than 1,000,000"),
-  dbPool: z
-    .number()
-    .min(0, 'Must be more than 0')
-    .max(1000, "Can't be more than 1000")
-    .optional()
-    .nullable(),
-})
+const formSchema = z
+  .object({
+    dbSchema: z.array(z.string()),
+    dbExtraSearchPath: z.string(),
+    maxRows: z.number().max(1000000, "Can't be more than 1,000,000"),
+    dbPool: z
+      .number()
+      .min(0, 'Must be more than 0')
+      .max(1000, "Can't be more than 1000")
+      .optional()
+      .nullable(),
+    enableDataApi: z.boolean(),
+  })
+  .refine(
+    (data) => {
+      if (data.enableDataApi && data.dbSchema.length === 0) {
+        return false
+      }
+      return true
```

Comment:
nit: can be simplified

```suggestion
      return !(data.enableDataApi && data.dbSchema.length === 0)
```

Reply:
not particular on this one - no worries!